tests/rptest/tests/pandaproxy_test.py

```diff
@@ -333,6 +334,132 @@ def _create_named_consumer(self,
+                "value"]
+            assert fetch_result_0[0]["partition"] == expected_data["records"][
+                0]["partition"]
+            assert fetch_result_0[0]["offset"] == expected_offset
+
+        def check_offsets(group_id: str,
+                          topic_name: str,
+                          expected_offset: int,
+                          auth_tuple: Optional[tuple[str, str]] = None,
+                          do_set_offsets: bool = True):
+            self.logger.debug(
+                f"Create a consumer and subscribe to topic: {topic_name}")
+            # A consumer is kept in a memory resident map within the kafka::client
+            # and that map is wiped after restart. Therefore, we create new consumer each time
+            cc_res = self._create_consumer(group_id, auth=auth_tuple)
+            assert cc_res.status_code == requests.codes.ok
+            c0 = Consumer(cc_res.json(), self.logger)
+            sc_res = c0.subscribe([topic_name], auth=auth_tuple)
+            assert sc_res.status_code == requests.codes.no_content
+
+            # Maybe set consumer offsets to 0
+            parts = [0, 1, 2]
+            if do_set_offsets:
+                sco_req = dict(partitions=[
+                    dict(topic=topic_name, partition=p, offset=expected_offset)
+                    for p in parts
+                ])
+                co_res_raw = c0.set_offsets(data=json.dumps(sco_req),
+                                            auth=auth_tuple)
+                assert co_res_raw.status_code == requests.codes.no_content
+
+            self.logger.debug(f"Check consumer offsets")
+            co_req = dict(partitions=[
+                dict(topic=topic_name, partition=p) for p in parts
+            ])
+            offset_result_raw = c0.get_offsets(data=json.dumps(co_req),
+                                               auth=auth_tuple)
+            assert offset_result_raw.status_code == requests.codes.ok
+            res = offset_result_raw.json()
+            # Should be one offset for each partition that we previously set offsets for
+            assert len(res["offsets"]) == len(parts)
+            for r in res["offsets"]:
+                assert r["topic"] == topic_name
+                assert r["partition"] in parts
+                assert r["offset"] == expected_offset
+                assert r["metadata"] == ""
+
+            # The consumer uses a gate to determine when new requests are serviced and removing
+            # the consumer safely closes the gate. If consumer remove is not called, then we risk
+            # an infinite loop of "gate closed exceptions" that lead to failures during RP broker
+            # shutdown. In ducktape, this will manifest as a MaxRetryError from the request lib.
+            self.logger.info("Remove consumer")
+            rc_res = c0.remove(auth=auth_tuple)
+            assert rc_res.status_code == requests.codes.no_content
```

Comment:
RP will hang/not-shutdown with many "gate closed exceptions" when someone fails to remove a consumer. This is a bug but it is unrelated to this PR which is about creating restart endpoints. I'll create a ticket for this and do some testing and more investigation into solutions. But I will also check with @BenPope about this next week because I may be doing something wrong as well.

Reply:
After investigating more, I believe the problem lies in this particular ducktape test but I do not have a solution yet. More details [here](https://github.com/redpanda-data/redpanda/issues/9310#issuecomment-1460710818)

My hypothesis is that there may be a leaking consumer or something after restarting the http-proxy which would indicate that I missed some state in the restart method.